src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,429 @@
+    try:
+        os.mkdir(path)
+    except FileExistsError:
+        pass
+
+    # To retrieve all the the deployements that are present in the Cluster
+    deployments_list = appv1_api_instance.list_namespaced_deployment("azure-arc")
+
+    # Traversing through all the deployments present
+    for deployment in deployments_list.items:
+
+        # Fetching the deployment name
+        deployment_name = deployment.metadata.name
+
+        # Creating a text file with the name of the deployment and adding deployment status in it
+        with open("C:\\Users\\t-svagadia\\diagnostic_logs\ " + time_stamp + "\Deployment_logs\ " + deployment_name + ".txt", 'w+') as deployment_file:
+            deployment_file.write(str(deployment.status))
+
+
+def check_agent_state(corev1_api_instance, time_stamp):
+
+    with open("C:\\Users\\t-svagadia\\diagnostic_logs\ " + time_stamp + "\Agent_State.txt", 'w+') as agent_state:
+
+        # To retrieve all of the arc agent pods that are presesnt in the Cluster
+        arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+        # Check if any arc agent other than kube aadp proxy is not in Running state
+        counter = 0
+        for each_agent_pod in arc_agents_pod_list.items:
+
+            # Storing the state of the arc agent in the user machine
+            agent_state.write(each_agent_pod.metadata.name + " = " + each_agent_pod.status.phase + "\n")
+
+            # If the agent is Kube add proxy we will continue with the next agent
+            if(each_agent_pod.metadata.name.startswith("kube-aad-proxy")):
+                continue
+            if each_agent_pod.status.phase != 'Running':
+                counter = 1
+
+        # Displaying error if the arc agents are in pending state.
+        if counter:
+            print("Error: One or more Azure Arc agents are in pending state. It may be caused due to insufficient resource availability on the cluster.\n For more details on resource requirement visit 'aka.ms\\arcenabledkubernetesresourcerequirement'. \n")
+            return False
+
+        return True
+
+
+def check_agent_version(connected_cluster, azure_arc_agent_version):
+
+    # If the agent version in the connected cluster resource is none skip the check
+    if(connected_cluster.agent_version is None):
+        return True
+
+    # To get user agent verison and the latest agent version
+    user_agent_version = connected_cluster.agent_version
+    current_user_version = user_agent_version.split('.')
+    latest_agent_version = azure_arc_agent_version.split('.')
+
+    # Comparing if the user version is comaptible or not
+    if((int(current_user_version[0]) < int(latest_agent_version[0])) or (int(latest_agent_version[1]) - int(current_user_version[1]) > 2)):
+        logger.warning("Error: We found that you are on an older agent version thats not supported.\n Please visit this link to know the agent version support policy 'link'.\n")
```

Comment:
Remove the "Error:" part in log statement since it's a warning

Reply:
it's not really an error right. We allow customers to disable auto-upgrade and hence they can be on an older version. We're just displaying the info that it is an older version which is not supported by our policy. But he can still stay on that old version.

Reply:
Ohk got it.